python/lib/core/dmod/core/dataset.py

```diff
@@ -125,6 +165,20 @@ class Dataset(Serializable):
     Note that it is not guaranteed that any such dataset still exist and/or are still available.""")
     created_on: Optional[datetime] = Field(description="When this dataset was created, or ``None`` if that is not known.")
     last_updated: Optional[datetime]
+    data_archiving: Optional[DataArchiving] = Field(default=None,
+                                                    description="Indication of what, if any, whole-dataset archiving "
+                                                                "(as opposed to simply containing an archive as one of "
+                                                                "the data files/items) is applied by dataset backend "
+                                                                "to store this dataset's data.")
+
+    @validator("data_archiving")
+    def validate_data_archiving(cls, v, values):
```

Comment:
I don't think we need to make `data_archiving` private, but it would be nicer if we could (easily) encapsulate the validation with a setter instead of it happening during init.

I have switched the validator itself to a root validator to make sure we avoid issues with defaults, etc..